app/plugins/Wallet/js/lifecycle.js

```diff
@@ -1,89 +1,163 @@
 'use strict';
 
+var blockheight = 0;
+
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Updating  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 // Make API calls, sending a channel name to listen for responses
-function update() {
+function update(address) {
 	IPC.sendToHost('api-call', '/wallet', 'update-status');
-	/*
+
+	// Get list of wallet addresses
 	IPC.sendToHost('api-call', {
-		url: '/wallet/transactions',
+		url: '/wallet/addresses',
+		type: 'GET',
+	}, 'update-address');
+
+	updating = setTimeout(update, 60000 * 5); // update every 5 min
+	return;
+}
+
+// Add transactions to view list per address
+addResultListener('update-address', function(result) {
+	// Update address list
+	eID('address-list').innerHTML = '';
+	result.addresses.forEach( function (address) {
+		appendAddress(address);
+	});
+	return;
+
+	/* Fetch all wallet transactions by iterate over wallet addresses
+	var loopmax = result.addresses.length;
+	var counter = 0;
+	(function next() {
+		setTimeout(function() {
+			updateAddrTxn(result.addresses[counter].address);
+			next();
+		}, 50); // force 50 ms delay between each GET request
+	})();*/
```

Comment:
What is this commented out stuff? A defined function that then gets executed? Why?